components/playlist/browser/resources/components/contextualMenu.tsx

```diff
@@ -126,27 +46,27 @@ export default function ContextualMenuAnchorButton ({
   onShowMenu,
   onDismissMenu
 }: MenuProps) {
-  const [showingMenu, setShowingMenu] = React.useState(false)
+  const menuButtonProps: ButtonMenuProps = {}
+  // Force menu widget to be closed when the anchor button is not visible. In
+  // case where it's visible, the menuButtonProps doesn't contain isOpen property,
+  // so it won't affect the behavior of the menu.
+  if (!visible) menuButtonProps.isOpen = false
+
+  // TODO(sko) We don't have event for opening menu widget. Once it's ready,
+  // wire onShowMenu and onDismissMenu to corresponding events.
   return (
-    <StyledAnchorButton visible={visible}>
-      <StyledLeoButtonContainer
-        onClick={e => {
-          e.stopPropagation()
-          setShowingMenu(true)
-          if (onShowMenu) onShowMenu()
-        }}
-      >
+    <StyledButtonMenu tabIndex={0} visible={visible} {...menuButtonProps}>
+      <div slot='anchor-content'>
         <Icon name='more-horizontal' />
-      </StyledLeoButtonContainer>
-      {showingMenu && (
-        <ContextualMenu
-          items={items}
-          onClickOutside={() => {
-            setShowingMenu(false)
-            if (onDismissMenu) onDismissMenu()
-          }}
-        />
-      )}
-    </StyledAnchorButton>
+      </div>
+      {items.map(i => (
+        <leo-menu-item key={i.name} onClick={i.onClick}>
```

Comment:
Note: You **might** be able to do 

```tsx
const StyledRow = styled('leo-menu-item')`
    foo: bar;
`
```

and not have a separate StyledRow and `leo-menu-item` here

Reply:
Nice! I didn't even try that 👍